### ISSUE TITLE
StackOverflowError when resolving method with duplicate segment in package name

### ISSUE THREAD

**Issue by xieziming** [javaparser/javaparser#2330]
code segment:
```
 MethodCallExpr mce = (MethodCallExpr)node;
 ResolvedMethodDeclaration rmd = mce.resolve();
```
It throws a StackOverflowError when running mce.resolve(). This issue happened when the node likes "
```
List<CreateOrModifyQualityResultParam> createQualityParamList = com.ali.com.google.common.collect.Lists.newArrayList();
```
When it trying to resolve type "com.ali.com.google.common.collect.Lists.newArrayList", it fall in to the dead loop of geting type of "com" and "com.ali" (the package name include 2 com)
```
at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getTypeConcrete(JavaParserFacade.java:448)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:310)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:292)
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.FieldAccessContext.solveSymbolAsValue(FieldAccessContext.java:74)
	at com.github.javaparser.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:68)
	at com.github.javaparser.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:73)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:205)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:44)
	at com.github.javaparser.ast.expr.FieldAccessExpr.accept(FieldAccessExpr.java:90)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getTypeConcrete(JavaParserFacade.java:448)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:310)
	at com.github.javaparser.symbolsolver.javaparsermodel.JavaParserFacade.getType(JavaParserFacade.java:292)
	at com.github.javaparser.symbolsolver.javaparsermodel.contexts.FieldAccessContext.solveSymbolAsValue(FieldAccessContext.java:74)
	at com.github.javaparser.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:68)
	at com.github.javaparser.symbolsolver.resolution.SymbolSolver.solveSymbolAsValue(SymbolSolver.java:73)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:277)
	at com.github.javaparser.symbolsolver.javaparsermodel.TypeExtractor.visit(TypeExtractor.java:44)
	at com.github.javaparser.ast.expr.NameExpr.accept(NameExpr.java:79)
```

**Comment by xieziming** [comment]
i believe a bug here

```
     // to resolve a name when in a fieldAccess context, we can get to the grand parent to prevent a infinite loop if the name is the same as the field (ie x.x)
                if (node.getParentNode().isPresent() && node.getParentNode().get() instanceof FieldAccessExpr && node.getParentNode().get().getParentNode().isPresent()) {
                    return getContext(node.getParentNode().get().getParentNode().get(), typeSolver);
                }
```

**Comment by matozoid** [comment]
Can you make a fix and open a PR?

**Comment by jlerbsc** [comment]
This issues seems to be resolved in JP 3.16.3

    public class Issue2330Test extends AbstractSymbolResolutionTest {

    @Test
    public void test() {
        final Path testRoot = adaptPath("src/test/resources/issue2330");
        TypeSolver reflectionTypeSolver = new ReflectionTypeSolver();
        JavaParserTypeSolver javaParserTypeSolver = new JavaParserTypeSolver(testRoot);
        CombinedTypeSolver combinedTypeSolver = new CombinedTypeSolver(reflectionTypeSolver, javaParserTypeSolver);
        ParserConfiguration configuration = new ParserConfiguration()
                .setSymbolResolver(new JavaSymbolSolver(combinedTypeSolver));
        StaticJavaParser.setConfiguration(configuration);

        String src = 
                "import a.b.a.A;\n" +
                "class X {\n" +
                "  public void f(){ " +
                "    A a = new A();\n" +
                "    a.m();\n" +
                "  }\n" + 
                "}";

        CompilationUnit cu = StaticJavaParser.parse(src);

        List<MethodCallExpr> mces = cu.findAll(MethodCallExpr.class);
        System.out.println(String.format("%s resolved to %s", mces.get(0).toString(),
                mces.get(0).resolve().getQualifiedSignature()));

    }
    }

with a class 

    package a.b.a;
    public class A
    {
      public void m() {   
      }
    }

result

    a.m() resolved to a.b.a.A.m()
